File: src/main/java/com/thebyteslayer/tracker/TrackerPlugin.java
package com.thebyteslayer.tracker;

import org.bukkit.plugin.java.JavaPlugin;

public class TrackerPlugin extends JavaPlugin {
    
    private CompassTracker compassTracker;
    
    @Override
    public void onEnable() {
        this.compassTracker = new CompassTracker(this);
        
        getServer().getPluginManager().registerEvents(new CompassListener(compassTracker), this);
        getServer().getPluginManager().registerEvents(new PlayerListener(compassTracker), this);
        getServer().getPluginManager().registerEvents(new PortalListener(compassTracker), this);
        

        
        getLogger().info("Tracker plugin has been enabled!");
    }
    
    @Override
    public void onDisable() {
        if (compassTracker != null) {
            compassTracker.cleanup();
        }
        getLogger().info("Tracker plugin has been disabled!");
    }
} 

File: src/main/kotlin/com/thebyteslayer/tracker/TrackerPlugin.kt
package com.thebyteslayer.tracker

import org.bukkit.plugin.java.JavaPlugin

class TrackerPlugin : JavaPlugin() {

    private var compassTracker: CompassTracker? = null

    override fun onEnable() {
        compassTracker = CompassTracker(this)

        server.pluginManager.registerEvents(CompassListener(compassTracker!!), this)
        server.pluginManager.registerEvents(PlayerListener(compassTracker!!), this)
        server.pluginManager.registerEvents(PortalListener(compassTracker!!), this)


        logger.info("Tracker plugin has been enabled!")
    }

    override fun onDisable() {
        compassTracker?.cleanup()
        logger.info("Tracker plugin has been disabled!")
    }
}
